File: codeforces/low_difficulty/difficulty_900/1.football_[96A]/football.py
# Author: José Rodolfo (jric2002)
p = str(input())
size_p = len(p)
dangerous = False
if (size_p >= 7):
  for i in range(0, (size_p - 7 + 1)):
    sub_p = p[i:(i + 7)]
    if (sub_p == "0000000" or sub_p == "1111111"):
      dangerous = True
      break
print("YES") if (dangerous) else print("NO")

File: codeforces/low_difficulty/difficulty_900/1.football_[96A]/Football.java
/* Author: José Rodolfo (jric2002) */
import java.util.Scanner;
public class Football {
  public static void main(String[] args) {
    String p, sub_p;
    int size_p;
    boolean dangerous;
    Scanner input = new Scanner(System.in);
    p = input.next();
    size_p = p.length();
    dangerous = false;
    if (size_p >= 7) {
      for (int i = 0; i < (size_p - 7 + 1); i++) {
        sub_p = p.substring(i, i + 7);
        if (sub_p.equals("0000000") || sub_p.equals("1111111")) {
          dangerous = true;
          break;
        }
      }
    }
    System.out.println((dangerous) ? "YES" : "NO");
  }
}